src/base/ceramicApiKit.ts

```diff
@@ -0,0 +1,142 @@
+      edges: ConfirmationEdge[];
+    };
+  }
+}
+
+export class ceramicApiKit {
+  private composeClient: any;
+  private privateKey: string;
+  private ceramicUrl: string;
+
+  constructor(privateKey: string, ceramicUrl: string) {
+    this.privateKey = privateKey;
+    this.ceramicUrl = ceramicUrl;
+  }
+
+  async connect() {
+    try {
+      const { ComposeClient } = await import('@composedb/client');
+      const composeClient = new ComposeClient({
+        ceramic: this.ceramicUrl,
+        //cannot import type from ESM only module in nest.js
+        // @ts-ignore
+        definition: definition,
+      });
+      const { fromString } = await import('uint8arrays/from-string');
+      const seedArray = fromString(this.privateKey, "base16");
+      const { Ed25519Provider } = await import('key-did-provider-ed25519');
+      const provider = new Ed25519Provider(new Uint8Array(seedArray));
+      const { DID } = await import('dids');
+      const { getResolver } = await import('key-did-resolver');
+      const did = new DID({ provider, resolver: getResolver() });
+      await did.authenticate();
+      composeClient.setDID(did);
+      this.composeClient = composeClient;
+    } catch (error) {
+      console.error('Error on connecting to Ceramic:', error);
+      throw error;
+    }
+  }
+
+  async proposeTransaction({
+                             safeAddress,
+                             safeTransactionData,
+                             safeTxHash,
+                             senderAddress,
+                             senderSignature,
+                             origin
+                           }: ProposeTransactionProps, threshold: number): Promise<void> {
+    if (!this.composeClient) {
+      await this.connect();
+    }
+    const confirmation = await this.composeClient.executeQuery(`
+            mutation CreateConfirmation {
+                createConfirmation(
+                    input: {
+                        content: {
+                            owner: "${senderAddress}"
+                            signature: "${senderSignature}"
+                            signatureType: "ECDSA"
+                            submissionDate: "${(new Date()).toISOString()}"
+                            transactionHash: "${safeTxHash}"
+                            confirmationType: "approve"
+                        }
+                        clientMutationId: null
+                    }
+                ) {
+                    clientMutationId
+                }
+            }
+        `);
+
+
+    return Promise.resolve();
+  }
+
+  async getTransactionioConfirmations(safeTxHash: string): Promise<SafeMultisigConfirmationResponse[]> {
+
+    try {
+      if (!this.composeClient) {
+        await this.connect();
+      }
+
+      const confirmationsIndex: ConfirmationIndexData = await this.composeClient.executeQuery(`
+            query ConfirmationIndex {
+                confirmationIndex(
+                    first: 99
```

Comment:
why this is 99, and does this number ensure that nothing goes wrong.

Reply:
just to higher the capacity, at the end of a day there's no absolute guarantee this query always works,
image someone is try to attack our service, and he sent 100 of fake signatures with the `safeTxHash`
which could not steal our money, but make the propose failed

personally, this hypothesis is low possibility to me, but set query limit to a reasonable large number makes me more confident about it running